File: Java/app/src/main/java/com/omeryildizce/java/Variables.java
package com.omeryildizce.java;

public class Variables {
    public static void main(String[] args) {

        // Variables
        int age = 24;
        System.out.println(age++);
        System.out.println(age / 5);
        System.out.println(++age);

        // Double - Float
        double z = 5.0;
        double a = 11.0;

        System.out.println(a/z);
        float f = 10f;

        double pi = 3.14;
        double mathPi = Math.PI;
        int radius = 5;
        System.out.println(2* pi * radius);
        System.out.println(2* mathPi * radius);

        String name = "James";
        String surname = "Hetfield";
        String fullname = name + " " + surname;
        String fullname1 = String.format("%s %s", name, surname);

        System.out.println(fullname1);

        // final
        int myInteger = 5;
        myInteger = 4;
        System.out.printf("my integer -> %d%n", myInteger);
        final int myFinalIteger = 5;
        // myFianlIteger = 8;
        System.out.printf("my final integer -> %d%n", myFinalIteger);

    }
}

File: s15_KotlinLearning/KotlinLearning/app/src/main/java/com/omeryildizce/kotlinlearning/kotlin/Variables.kt
package com.omeryildizce.kotlinlearning

import java.util.*

fun main() {
    // Integer
    // varInteger()

    // varDouble
    // varDouble()

    // String
    // varString()

    // boolean
    // varBoolean()

    // casting
    cast()
}

fun cast() {
    var myNumber = 5
    var myLongNumber = myNumber.toLong()

    var input = "10"
    var inputInteger = input.toInt()

    println(inputInteger * 2)


}

fun varBoolean() {
    var myBoolean = true
    myBoolean = false

    /**
     * <
     * >
     * <=
     * >=
     * ==
     * !=
     * &&
     * ||
     */

    println(3 < 5)
    println(6 < 5)
    println(3 == 3)
    println(3 != 3)



}

fun varString() {
    val myString = "Ömer"
    val name ="Ömer"
    val surname = "YILDIZ"
    val fullName: String = name + " "+ surname
    println(fullName)
    myString.lowercase()
    myString.replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.ROOT) else it.toString() }
    println(myString)
}


fun varDouble() {
    val pi = Math.PI
    println(pi)

   val myFloat = 3.14f
    println(myFloat)

    var myAge:Double
    myAge = 23.0
    println(myAge / 2)

}

private fun varInteger() {
    var x = 5
    var y = 4
    println(x * y)

    var age = 35
    println(age / 7 * 5)
    val result = age / 7 * 5
    println(result)
    age = 36
    println(age)

    val a = 5
    val myInteger: Int
    myInteger = 10

    var myLong : Long = 10
}
